signac/testing.py

```diff
@@ -5,6 +5,8 @@
 from itertools import cycle
 
 
+# TODO: This feels like something that should be a test fixture or
+# fixture-adjacent, not code that belongs in the main package.
```

Comment:
Tend to disagree on this one. There is value in user-facing code to easily generate a sample project. The fixture code would necessarily be shipped with the package.

It's also used here: https://github.com/glotzerlab/signac-flow/blob/4d49e4f72e8d2201fa27efe2fbb3b0940a31ca79/flow/testing.py#L35

Reply:
I agree in principle. In practice, I don't know if anyone has ever used this. I certainly haven't. Perhaps we should be doing a better job publicizing its existence if we think it could help users with prototyping something?

Reply:
I have always wanted to replicate this kind of thing (shipping helpful test fixtures) in more packages I work on, but have never gotten around to it. I believe I tried using this feature in the benchmarking code, but it's not meant to generate large projects suitable for benchmarking -- only small projects with complex schemas. We might want to revisit that at some point, and try to unify those two fixture-like code paths.

Reply:
It looks like this function is undocumented.

Reply:
@bdice I'm not sure I fully understand where you fall on this based on your comments. Are you recommending to keep this, but maybe document and improve? Or would you rather remove it?

Reply:
I think it’s a nice feature. Keep it, document it, and improve it.

Reply:
Ok, ideally we use these exact functions as part of the pytest fixtures.

Reply:
OK I'll drop this in this PR and make an issue to improve this.

Reply:
Consider linking the issue in the TODO, perhaps.